datastore/query.go

```diff
@@ -176,6 +323,22 @@ func (q *Query) Transaction(t *Transaction) *Query {
 	return q
 }
 
+// FilterEntity returns a query with provided filter.
+//
+// Filter can be a single field comparison or a composite filter
+// AND and OR are supported composite filters
+// Filters in multiple calls are joined together by AND
+func (q *Query) FilterEntity(entityFilter EntityFilter) *Query {
+	q = q.clone()
+	vf, err := entityFilter.toEntityFilter()
```

Comment:
nit: The naming seems a little circular here. Design looks good, but I wonder if there's a better name.

Reply:
Changed the name to toValidFilter